src/melee/it/types.h

```diff
@@ -44,40 +58,75 @@
 #define ITEM_UNK_LOCKON 5
 #define ITEM_UNK_ENEMY 6 // Item type: Stage Enemy (Goomba, Koopa Troopa, etc.)
 
+/// @todo Size unknown.
 struct ItemStateTable {
-    s32 x0_state_id;
-    void* x4_callback_anim;
-    void* x8_callback_phys;
-    void* xC_callback_coll;
+    /// @at{0} @sz{4}
+    enum_t asid;
+
+    /// @at{4} @sz{4}
+    HSD_GObjPredicate animated;
+
+    /// @at{8} @sz{4}
+    HSD_GObjEvent physics_updated;
+
+    /// @at{C} @sz{4}
+    HSD_GObjPredicate collided;
 };
 
+/// @todo Size unknown.
 struct ItemStateContainer {
-    struct ItemStateTable stateTable[
-#ifdef M2CTX
-        0
-#endif
-    ];
+    /// @at{0}
+    struct ItemStateTable stateTable UNK_SIZE_ARRAY;
 };
 
 struct ItemLogicTable {
-    struct ItemStateContainer* x0_itemStatePtr;
-    void (*x4_callback_OnSpawn)(HSD_GObj* item);
-    void (*x8_callback_OnDestroy)(HSD_GObj* item);
-    void (*xC_callback_OnPickup)(HSD_GObj* item);
-    void (*x10_callback_OnDrop)(HSD_GObj* item);
-    void (*x14_callback_OnThrow)(HSD_GObj* item);
-    bool (*x18_callback_OnGiveDamage)(HSD_GObj* item);
-    bool (*x1C_callback_OnTakeDamage)(HSD_GObj* item);
-    void (*x20_callback_EnterAir)(HSD_GObj* item);
-    bool (*x24_callback_OnReflect)(HSD_GObj* item);
-    bool (*x28_callback_OnClank)(HSD_GObj* item);
-    bool (*x2C_callback_OnAbsorb)(HSD_GObj* item);
-    bool (*x30_callback_OnShieldBounce)(HSD_GObj* item);
-    bool (*x34_callback_OnHitShield)(HSD_GObj* item);
-    void (*x38_callback_OnUnknown)(HSD_GObj* item, HSD_GObj* fighter);
-};
-
-typedef struct _CameraBoxFlags {
+    /// @at{0} @sz{4}
+    struct ItemStateContainer* states;
+
+    /// @at{4} @sz{4}
+    void (*spawned)(HSD_GObj* item);
+
+    /// @at{8} @sz{4}
+    void (*destroyed)(HSD_GObj* item);
+
+    /// @at{C} @sz{4}
+    void (*picked_up)(HSD_GObj* item);
```

Comment:
```suggestion
    void (*destroyed)(HSD_GObj* item); /// @at{8} @sz{4}
    void (*picked_up)(HSD_GObj* item); /// @at{C} @sz{4}
```

Can we do something like this for the structs? They're taking up so much vertical space, makes it way harder to get a gist of them

Reply:
using a different struct as an example because it's small. Options are...

1. The current way
```c
/// @sz{1C}
struct Item_DynamicBones {
    /// @at{0} @sz{4}
    /// @todo If this is 256, dynamics are not processed
    int flags;

    /// @at{4} @sz{4}
    HSD_JObj* skeleton;

    /// @at{8} @sz{4}
    /// @todo Stored at @c 8000FDD4, comes from a nonstandard heap at
    ///       @c -0x52fc(r13)
    unk_t unk_ptr;

    /// @at{C} @sz{4}
    /// @brief The number of bones in this bone set.
    int count;

    /// @at{10} @sz{C}
    Vec3 unk_vec;
};
```

2. What you suggested, annotations to the right. I personally hate it because it requires you to parse variable length lines of text just to see the offsets (when looking at the source), also looks pretty claustrophobic. Note that the `<` is required by Doxygen.
```c
/// @sz{1C}
struct Item_DynamicBones {

    /// @todo If this is 256, dynamics are not processed
    int flags; ///< @at{0} @sz{4}

    HSD_JObj* skeleton; ///< @at{4} @sz{4}

    /// @todo Stored at @c 8000FDD4, comes from a nonstandard heap at
    ///       @c -0x52fc(r13)
    unk_t unk_ptr; ///< @at{8} @sz{4}
    
    /// @brief The number of bones in this bone set.
    int count; ///< @at{C} @sz{4}

    /// @at{10} @sz{C}
    Vec3 unk_vec;
};
```

3. Annotations on the left. Looks pretty awful imo, and if the struct goes into multi-digit addresses then it's not clear where the space should go.
```c
/// @sz{1C}
struct Item_DynamicBones {
    /// @todo If this is 256, dynamics are not processed
    /** @at{8} @sz{4} */ int flags;

    /** @at{8} @sz{4} */ HSD_JObj* skeleton;

    /// @todo Stored at @c 8000FDD4, comes from a nonstandard heap at
    ///       @c -0x52fc(r13)
    /** @at{8} @sz{4} */ unk_t unk_ptr;

    /// @brief The number of bones in this bone set.
    /** @at{C} @sz{4} */ int count;

    /** @at{10} @sz{C} */ Vec3 unk_vec;
};
```

Those are the most vertically compact options, I think. What's the use case when you say "harder to get a gist of them"? The docs summary looks like this
![image](https://user-images.githubusercontent.com/5190964/212459940-701ce90f-ef97-462a-a4b7-bf94ef95c51a.png)

Reply:
Keep in mind that the goal is to eventually have at least one sentence per member, often full paragraphs, at which point scanning the structs will be very unpleasant no matter how it's formatted.

Reply:
Option 4: annotate structs in separate markdown files.

Reply:
Which would look something like this, copypasted from stackoverflow
```c
/** @struct foreignstruct
 *  @brief This structure blah blah blah...
 *  @var foreignstruct::a 
 *  Member 'a' contains...
 *  @var foreignstruct::b 
 *  Member 'b' contains...
 */
```

Reply:
That's fair, once these are fully populated with info it'll make more sense